### PR TITLE
etcdmain: deprecate --ca-file and --peer-ca-file

### PR DESCRIPTION
**Pull request by yichengq**
1. Print out DEPRECATE warning when people are still using it.
2. Remove it from help message.

for #2609 
/cc @kelseyhightower @barakmich

### CODE REVIEW

**Comment by yichengq** [comment]
ping @kelseyhightower Please help me on reviewing the doc change.

File: Documentation/configuration.md [hunk 1]
```diff
@@ -109,10 +109,6 @@ To start etcd automatically using custom settings at startup in Linux, using a [
 
 The security flags help to [build a secure etcd cluster][security].
 
-##### -ca-file
```
Comment (kelseyhightower) [inline Documentation/configuration.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 2]
```diff
@@ -18,7 +18,9 @@ etcd takes several certificate related configuration options, either through com
 
 `--key-file=<path>`: Key for the certificate. Must be unencrypted.
 
-`--ca-file=<path>`: When this is set etcd will check all incoming HTTPS requests for a client certificate signed by the supplied CA, requests that don't supply a valid client certificate will fail.
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 3]
```diff
@@ -28,7 +30,9 @@ The peer options work the same way as the client-to-server options:
 
 `--peer-key-file=<path>`: Key for the certificate. Must be unencrypted.
 
-`--peer-ca-file=<path>`: When set, etcd will check all incoming peer requests from the cluster for valid client certificates signed by the supplied CA.
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 4]
```diff
@@ -68,12 +72,10 @@ You need the same files mentioned in the first example for this, as well as a ke
 
 ```sh
 $ etcd -name infra0 -data-dir infra0 \
-  -ca-file=/path/to/ca.crt -cert-file=/path/to/server.crt -key-file=/path/to/server.key \
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 5]
```diff
@@ -68,12 +72,10 @@ You need the same files mentioned in the first example for this, as well as a ke
...
+  -client-cert-auth -trusted-ca-file=/path/to/ca.crt -cert-file=/path/to/server.crt -key-file=/path/to/server.key \
   -advertise-client-urls https://127.0.0.1:2379 -listen-client-urls https://127.0.0.1:2379
 ```
 
-Notice that the addition of the `-ca-file` option automatically enables client certificate checking.
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 6]
```diff
@@ -130,13 +132,13 @@ DISCOVERY_URL=... # from https://discovery.etcd.io/new
 
 # member1
 $ etcd -name infra1 -data-dir infra1 \
-  -peer-ca-file=/path/to/ca.crt -peer-cert-file=/path/to/member1.crt -peer-key-file=/path/to/member1.key \
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: Documentation/security.md [hunk 7]
```diff
@@ -130,13 +132,13 @@ DISCOVERY_URL=... # from https://discovery.etcd.io/new
...
+  -peer-client-cert-auth -peer-trusted-ca-file=/path/to/ca.crt -peer-cert-file=/path/to/member1.crt -peer-key-file=/path/to/member1.key \
   -initial-advertise-peer-urls=https://10.0.1.10:2380 -listen-peer-urls=https://10.0.1.10:2380 \
   -discovery ${DISCOVERY_URL}
 
 # member2
 $ etcd -name infra2 -data-dir infra2 \
-  -peer-ca-file=/path/to/ca.crt -peer-cert-file=/path/to/member2.crt -peer-key-file=/path/to/member2.key \
```
Comment (kelseyhightower) [inline Documentation/security.md]: I think we need to keep the docs around, just mark them deprecated

File: etcdmain/config.go [hunk 8]
```diff
@@ -171,12 +171,12 @@ func NewConfig() *config {
 	}
 
 	// security
-	fs.StringVar(&cfg.clientTLSInfo.CAFile, "ca-file", "", "Path to the client server TLS CA file.")
+	fs.StringVar(&cfg.clientTLSInfo.CAFile, "ca-file", "", "DEPRECATED: Path to the client server TLS CA file.")
```
Comment (kelseyhightower) [inline etcdmain/config.go]: right.

File: etcdmain/config.go [hunk 9]
```diff
@@ -171,12 +171,12 @@ func NewConfig() *config {
...
 	fs.StringVar(&cfg.clientTLSInfo.CertFile, "cert-file", "", "Path to the client server TLS cert file.")
 	fs.StringVar(&cfg.clientTLSInfo.KeyFile, "key-file", "", "Path to the client server TLS key file.")
 	fs.BoolVar(&cfg.clientTLSInfo.ClientCertAuth, "client-cert-auth", false, "Enable client cert authentication.")
 	fs.StringVar(&cfg.clientTLSInfo.TrustedCAFile, "trusted-ca-file", "", "Path to the client server TLS trusted CA key file.")
-	fs.StringVar(&cfg.peerTLSInfo.CAFile, "peer-ca-file", "", "Path to the peer server TLS CA file.")
+	fs.StringVar(&cfg.peerTLSInfo.CAFile, "peer-ca-file", "", "DEPRECATED: Path to the peer server TLS CA file.")
```
Comment (kelseyhightower) [inline etcdmain/config.go]: right.

File: etcdmain/help.go [hunk 10]
```diff
@@ -75,8 +75,6 @@ proxy flags:
 
 security flags:
 
-	--ca-file ''
-		path to the client server TLS CA file.
```
Comment (kelseyhightower) [inline etcdmain/help.go]: I think we need to keep the docs around, just mark them deprecated

File: etcdmain/help.go [hunk 11]
```diff
@@ -85,8 +83,6 @@ security flags:
 		enable client cert authentication.
 	--trusted-ca-file ''
 		path to the client server TLS trusted CA key file.
-	--peer-ca-file ''
```
Comment (kelseyhightower) [inline etcdmain/help.go]: I think we need to keep the docs around, just mark them deprecated

**Comment by kelseyhightower** [comment]
lgtm

**Pull request merged**